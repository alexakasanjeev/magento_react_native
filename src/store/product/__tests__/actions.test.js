import {
  addToCart,
  getProductMedia,
  setCurrentProduct,
  resetAddToCartState,
  getConfigurableChildren,
  getConfigurableProductOptions,
} from '../actions';

describe('product actions', () => {
  test('setCurrentProduct()', () => {
    // Setup
    const productType = 'simple';
    const sku = 'WH02-S';
    const children = null;

    // Exercise
    const result = setCurrentProduct(productType, sku, children);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('getConfigurableProductOptions()', () => {
    // Setup
    const sku = 'WH02-S';

    // Exercise
    const result = getConfigurableProductOptions(sku);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('getConfigurableChildren()', () => {
    // Setup
    const sku = 'WH02-S';

    // Exercise
    const result = getConfigurableChildren(sku);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('getProductMedia()', () => {
    // Setup
    const sku = 'WH02-S';

    // Exercise
    const result = getProductMedia(sku);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('addToCart()', () => {
    // Setup
    const cartItem = {
      sku: 'WH02-S',
      qty: 1,
      quote_id: 23,
    };

    // Exercise
    const result = addToCart(cartItem);

    // Verify
    expect(result).toMatchSnapshot();
  });

  test('resetAddToCartState()', () => {
    // Setup
    const sku = 'WH02-S';

    // Exercise
    const result = resetAddToCartState(sku);

    // Verify
    expect(result).toMatchSnapshot();
  });
});
