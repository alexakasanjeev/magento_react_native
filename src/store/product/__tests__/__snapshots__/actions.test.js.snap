// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`product actions addToCart() 1`] = `
Object {
  "payload": Object {
    "cartItem": Object {
      "qty": 1,
      "quote_id": 23,
      "sku": "WH02-S",
    },
  },
  "type": "MAGENTO_ADD_TO_CART_REQUEST",
}
`;

exports[`product actions getConfigurableChildren() 1`] = `
Object {
  "payload": Object {
    "sku": "WH02-S",
  },
  "type": "MAGENTO_CONFIGURABLE_CHILDREN_REQUEST",
}
`;

exports[`product actions getConfigurableProductOptions() 1`] = `
Object {
  "payload": Object {
    "sku": "WH02-S",
  },
  "type": "MAGENTO_CONF_OPTIONS_REQUEST",
}
`;

exports[`product actions getProductMedia() 1`] = `
Object {
  "payload": Object {
    "sku": "WH02-S",
  },
  "type": "MAGENTO_PRODUCT_MEDIA_REQUEST",
}
`;

exports[`product actions resetAddToCartState() 1`] = `
Object {
  "payload": Object {
    "sku": "WH02-S",
  },
  "type": "RESET_ADD_TO_CART_STATE",
}
`;

exports[`product actions setCurrentProduct() 1`] = `
Object {
  "payload": Object {
    "children": null,
    "productType": "simple",
    "sku": "WH02-S",
  },
  "type": "UI_OPEN_SELECTED_PRODUCT_REQUEST",
}
`;
